test/e2e/specs/editor/various/inserting-blocks.spec.js

```diff
@@ -386,6 +386,320 @@ test.describe( 'Inserting blocks (@firefox, @webkit)', () => {
 
 		await expect( blockLibrary ).toBeHidden();
 	} );
+
+	test( 'should insert block with the slash inserter when using multiple words', async ( {
+		admin,
+		editor,
+		page,
+	} ) => {
+		await admin.createNewPost();
+		await page.keyboard.press( 'Enter' );
+		await page.keyboard.type( '/tag cloud' );
+
+		await expect(
+			page.getByRole( 'option', { name: 'Tag Cloud', selected: true } )
+		).toBeVisible();
+		await page.keyboard.press( 'Enter' );
+
+		await expect(
+			editor.canvas.getByRole( 'document', { name: 'Block: Tag Cloud' } )
+		).toBeVisible();
+	} );
+
+	// Check for regression of https://github.com/WordPress/gutenberg/issues/23263
+	test( 'inserts blocks at root level when using the root appender while selection is in an inner block', async ( {
+		admin,
+		editor,
+		page,
+	} ) => {
+		await admin.createNewPost();
+		await editor.insertBlock( { name: 'core/buttons' } );
+
+		// After inserting the Buttons block the inner button block should be selected.
+		await expect(
+			editor.canvas.getByRole( 'textbox', { name: 'Button text' } )
+		).toBeFocused();
+		await page.keyboard.type( '1.1' );
+
+		// The block appender is only visible when there's no selection.
+		await page.evaluate( () =>
+			window.wp.data.dispatch( 'core/block-editor' ).clearSelectedBlock()
+		);
+
+		// Insert a new block using the root appender.
+		await editor.canvas
+			.getByRole( 'button', { name: 'Add block' } )
+			.click();
+		await page
+			.getByRole( 'searchbox', {
+				name: 'Search for blocks and patterns',
+			} )
+			.fill( 'Paragraph' );
+		await page.getByRole( 'option', { name: 'Paragraph' } ).click();
+		await page.keyboard.type( '2' );
+
+		// Should show a buttons block followed by a paragraph.
+		await expect
+			.poll( editor.getBlocks )
+			.toMatchObject( [
+				{ name: 'core/buttons' },
+				{ name: 'core/paragraph' },
+			] );
+	} );
+
+	// // Check for regression of https://github.com/WordPress/gutenberg/issues/24262.
+	test( 'inserts a block in proper place after having clicked `Browse All` from inline inserter', async ( {
+		admin,
+		editor,
+		page,
+	} ) => {
+		await admin.createNewPost();
+		await page.keyboard.press( 'Enter' );
+		await page.keyboard.type( 'First paragraph' );
+		await page.keyboard.press( 'Enter' );
+		await page.keyboard.type( '## Heading' );
+		await page.keyboard.press( 'Enter' );
+		await page.keyboard.type( 'Second paragraph' );
+		await page.keyboard.press( 'Enter' );
+		await page.keyboard.type( 'Third paragraph' );
+
+		const boundingBox = await editor.canvas
+			.getByRole( 'document', { name: 'Block: Heading' } )
+			.boundingBox();
+
+		// Using the between inserter.
+		await page.mouse.move(
+			boundingBox.x + boundingBox.width / 2,
+			boundingBox.y - 10,
+			{ steps: 10 }
```

Comment:
@kevin940726, do you remember why we need to set an arbitrary number of steps here? The action doesn't work without this option, but I want to provide an inline comment here.

Reply:
From what I know, the testing environment/framework doesn't have the concept of a proper cursor that smoothly moves around so it has to simulate it.

If you move from point A to point B, it interpolates 10 positions between those points, and then the cursor will sort of 'appear' at those points to simulate moving, and that way it triggers things like hovering in and out of elements along the way.

10 is a bit of an arbitrary number, but I guess it worked well 😄

Reply:
Oops, did I not include comments? 🤦 Yep exactly what @talldan said ☝️.

IIRC, it's also related to how the insertion logic works. It first needs a `mouseenter` event on the canvas to toggle some flags, then finally it can read the `mousemove` event based on the position to trigger the blue line. We can argue that it's a potential bug for non-pointer devices though.

Reply:
Thank you, both. I’ll add inline comment.